lib/core/utils/select.js

```diff
@@ -85,3 +81,19 @@ function select(selector, context) {
 }
 
 export default select;
+
+/**
+ * Return a filter method to test if a node is in context; or
+ *  null if the node is always included.
+ * @private
+ * @param {Context}
+ * @return {Function|null}
+ */
+function getContextFilter(context) {
```

Comment:
I have no idea what tests to write for this. I'm not sure it's necessary, since PR doesn't change output. It just makes some things faster. If you feel tests are necessary, please let me know how.

Reply:
You could create a very deep shadow tree that times out the tests with the old code and doesn't now?

Reply:
I suppose so... Didn't try that. I'll play around with that for a bit, see if that works.